creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -150,14 +66,119 @@ const SecondarySyncHealthTracker = {
    * Builds redis key pattern given params
    * Key pattern string can map to one or multiple keys
    */
-  _getRedisKeyPattern (secondary, wallet, syncType, outcome, date = null) {
+  _getRedisKeyPattern ({ secondary = '*', wallet = '*', syncType = '*', outcome = '*', date = null }) {
     const prefix = 'SecondarySyncRequestOutcomes-Daily'
 
     // format: YYYY-MM-DD
     date = date || new Date().toISOString().split('T')[0]
 
     return `${prefix}:::${secondary}:::${wallet}:::${syncType}:::${date}:::${outcome}`
+  },
+
+  _parseSecondaryFromRedisKey (key) {
+    return key.split(':::')[1]
+  }
+}
+
+const Setters = {
+  async recordSuccess (secondary, wallet, syncType) {
+    await Setters._recordSyncRequestOutcome(secondary, wallet, syncType, true)
+  },
+
+  async recordFailure (secondary, wallet, syncType) {
+    await Setters._recordSyncRequestOutcome(secondary, wallet, syncType, false)
+  },
+
+  async _recordSyncRequestOutcome (secondary, wallet, syncType, success = true) {
```

Comment:
nit - this is a setter but in this structure maybe better to put in Utils?

Reply:
fair, done